opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,176 @@
+            is meter_name
+            is meter_version
+            is meter_schema_url
+            is None
+        ):
+            raise Exception(
+                "View name or some instrument selection "
+                "criteria must be provided for View {name}"
+            )
+
+        self._instrument_type = instrument_type
+        self._instrument_name = instrument_name
+        self._meter_name = meter_name
+        self._meter_version = meter_version
+        self._meter_schema_url = meter_schema_url
+
+        self._name = name
+
+        if self._name is not None:
+            if self._instrument_type is not None:
+                raise Exception(
+                    "Instrument type matching disabled "
+                    "for view with a defined name"
+                )
+
+            if self._instrument_name is not None:
+                raise Exception(
+                    "Instrument name matching disabled "
+                    "for view with a defined name"
+                )
+
+            if self._meter_name is not None:
+                raise Exception(
+                    "Instrument meter name matching disabled "
+                    "for view with a defined name"
+                )
+
+            if self._meter_version is not None:
+                raise Exception(
+                    "Instrument meter version matching disabled "
+                    "for view with a defined version"
+                )
+
+            if self._meter_schema_url is not None:
+                raise Exception(
+                    "Instrument meter schema URL matching disabled "
+                    "for view with a defined version"
+                )
+
+        self._description = description
+        self._attribute_keys = attribute_keys
+        self._aggregation = aggregation
+
+        self._has_name_and_has_matched = False
+
+    # pylint: disable=too-many-return-statements
+    def _match(self, instrument: Instrument) -> bool:
+
+        if self._has_name_and_has_matched:
+            return False
+
+        if self._instrument_type is not None:
+            if not isinstance(instrument, self._instrument_type):
+                return False
+
+        if self._instrument_name is not None:
+            if self._instrument_name != instrument.name:
+                return False
+
+        if self._meter_name is not None:
+            # pylint: disable=protected-access
+            if self._meter_name != instrument._meter.name:
+                return False
+
+        if self._meter_version is not None:
+            # pylint: disable=protected-access
+            if self._meter_version != instrument._meter.version:
+                return False
+
+        if self._meter_schema_url is not None:
+            # pylint: disable=protected-access
+            if self._meter_schema_url != instrument._meter.schema_url:
+                return False
+
+        if self._name is not None:
+            self._has_name_and_has_matched = True
```

Comment:
@ocelotl as https://github.com/open-telemetry/opentelemetry-specification/issues/2337 is being addressed, instruments may match once per Meter. I think we are better off tracking this in the SDK internals, wdyt?

Reply:
I think I found a way to solve this directly in the view, please take a look :v: